llvm/lib/Transforms/Vectorize/VPlanTransforms.cpp

```diff
@@ -913,9 +913,14 @@ static void simplifyRecipe(VPRecipeBase &R, VPTypeAnalysis &TypeInfo) {
         unsigned ExtOpcode = match(R.getOperand(0), m_SExt(m_VPValue()))
                                  ? Instruction::SExt
                                  : Instruction::ZExt;
-        auto *VPC =
-            new VPWidenCastRecipe(Instruction::CastOps(ExtOpcode), A, TruncTy);
-        VPC->insertBefore(&R);
+        VPValue *VPC;
+        if (auto *UV = R.getOperand(0)->getUnderlyingValue())
```

Comment:
nit: couldn't you just set UV to `nullptr`? Or return `nullptr` from `getUnderlyingValue`?

Then this would just be a single call. It took me a second pass to parse the semantics here.